instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/mongo/MongoClientInstrumentationAutoConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.autoconfigure.instrumentation.mongo;
+
+import com.mongodb.MongoClientSettings;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.mongo.v3_1.MongoTelemetry;
+import io.opentelemetry.instrumentation.spring.autoconfigure.internal.SdkEnabled;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.mongo.MongoClientSettingsBuilderCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+
+@ConditionalOnBean(OpenTelemetry.class)
+@ConditionalOnClass(MongoClientSettings.class)
+@ConditionalOnProperty(name = "otel.instrumentation.mongo.enabled", matchIfMissing = true)
+@Conditional(SdkEnabled.class)
+@Configuration
+public class MongoClientInstrumentationAutoConfiguration {
+
+  @Bean
+  public MongoClientSettingsBuilderCustomizer customizer(
+      OpenTelemetry openTelemetry,
+      @Value("${otel.instrumentation.mongo.statement-sanitizer.enabled:true}")
```

Comment:
this is not compatible with spring boot 2 - see https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/11263